Project/Source/Components/ComponentAnimation.cpp

```diff
@@ -1,21 +1,69 @@
 #include "ComponentAnimation.h"
 
 #include "Application.h"
+#include "Transition.h"
 #include "GameObject.h"
+#include "AnimationInterpolation.h"
 #include "AnimationController.h"
 #include "Resources/ResourceAnimation.h"
+#include "Resources/ResourceClip.h"
 #include "Components/ComponentType.h"
 #include "Components/ComponentTransform.h"
 #include "Modules/ModuleEditor.h"
 #include "Modules/ModuleResources.h"
+#include "Modules/ModuleTime.h"
+#include "Modules/ModuleInput.h"
+#include "Utils/UID.h"
+#include "Utils/Logging.h"
+
+#include <algorithm> // std::find
 
 #include "Utils/Leaks.h"
 
 #define JSON_TAG_LOOP "Controller"
 #define JSON_TAG_ANIMATION_ID "AnimationId"
+#define JSON_TAG_STATE_MACHINE_ID "StateMachineId"
+#define JSON_TAG_INITAL_STATE_ID "InitalState"
+#define JSON_TAG_CLIP "Clip"
 
 void ComponentAnimation::Update() {
-	animationController.Update();
+	// TODO remove, HardCoded for test transitions
+	/*
+	if (App->input->GetKey(SDL_SCANCODE_1)) {
+		if (t != 1) {
```

Comment:
Deprecated code?

Reply:
We have this in case we need to test the animation in cases that it is breaks , we want to delete this once it has passed the vertical slide. But if needed we could delete and retrieve this from the commits